client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -453,37 +445,53 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 					verticalId: siteVerticalId,
 				} ).then( () => reduxDispatch( requestActiveTheme( site?.ID || -1 ) ) )
 			);
-			recordTracksEvent( 'calypso_signup_select_design', {
-				...getEventPropsByDesign( _selectedDesign, selectedStyleVariation ),
-				...( positionIndex >= 0 && { position_index: positionIndex } ),
-				device: resolveDeviceTypeByViewPort(),
-			} );
 
-			if ( _selectedDesign.verticalizable ) {
-				recordTracksEvent(
-					'calypso_signup_select_verticalized_design',
-					getEventPropsByDesign( _selectedDesign, selectedStyleVariation )
-				);
-			}
+			handleSubmit(
+				{
+					selectedDesign: _selectedDesign,
+					selectedSiteCategory: categorization.selection,
+				},
+				{ ...( positionIndex >= 0 && { position_index: positionIndex } ) }
+			);
+		}
+	}
 
-			handleSubmit( {
-				selectedDesign: _selectedDesign,
-				selectedSiteCategory: categorization.selection,
+	function pickBlankCanvasDesign( blankCanvasDesign: Design, shouldGoToAssemblerStep: boolean ) {
+		if ( shouldGoToAssemblerStep ) {
+			setIntent( SiteIntent.Assembler );
+			setSelectedDesign( {
+				...blankCanvasDesign,
+				design_type: 'assembler',
 			} );
+		} else {
+			pickDesign( blankCanvasDesign );
 		}
 	}
 
-	function handleSubmit( providedDependencies?: ProvidedDependencies ) {
-		const _selectedDesign = providedDependencies?.selectedDesign as Design;
+	useEffect( () => {
```

Comment:
Why do we need this effect to submit the step? Instead, we can handle this inside `pickBlankCanvasDesign`, and then we don't need to wait for the component updated. If the reason is we cannot get the correct intent, we can just pass it as part of the `providedDependencies`. What do you think?

Reply:
"Fixed"!